### ISSUE TITLE
Permit detecting and importing a CTDL-ASN file from a remote URL.

### ISSUE THREAD

**Issue by Lomilar** [cassproject/cass-editor#601]

**Comment by Lomilar** [comment]
Should:

- [ ] Download the file from the remote endpoint and depending on what it is, invoke the standard file import screens.
- [ ] Detect when a CORS error prevents download and notify the person that they may need to download the data and import it via a file.

**Comment by Lomilar** [comment]
Options here:

Tell the server to import and own/manage the CTDL/ASN file from the remote URL (avoids CORS).  
OR  
Get the files up on the web and not behind CORS.

**Comment by woodkri** [comment]
I see now where the date formats came from with the O*NET files. The CTDL-ASN export from CaSS has the dates in the correct format, but if you go directly to a /resources/ or /graph/ URL, the dates are in this format: 

```ceasn:dateCreated: "07/02/2019 15:20:08"```

So somewhere in the publish process the date format is being changed, which will cause problems when trying to import these frameworks. They appear to import but the search can't find them.

This kind of URL (https://cass.credentialengine.org/api/ceasn/...) works since the dates are in the expected format.

Solutions: Either 1. Have the publish process keep the dates in the format exported from CaSS, or 2. Have the import check the format of the date fields and change if necessary. I think 1 is the better option since the fields are defined as xsd:date and xsd:dateTime.

**Comment by cwd-mparsons** [comment]
@woodkri 
The date issue stems from the library that we were using for JSON serialization: Newtonsoft:
Json.NET interprets and modifies ISO dates when deserializing to JObject
https://github.com/JamesNK/Newtonsoft.Json/issues/862

I found that Microsoft had recently added a new library: System.Text.Json
I started to use the latter for publishing competency frameworks - as a test/prototype. I had to make many changes for initial testing. I wanted to hold off making wholesale changes until happy that the serialization works properly without additional issue. I have updated the test and staging servers with the new version that properly retains the date format, as shown below:

https://sandbox.credentialengineregistry.org/graph/ce-21f4e17d-9189-4feb-b732-055fa3288dea
![image](https://user-images.githubusercontent.com/26332112/68243097-693a7100-ffd7-11e9-871f-ad5dbe7f1f07.png)

So the latter appears promising. 
After a little more testing, we can promote this to production, and I will make the other necessary changes.
